parlai/crowdsourcing/utils/analysis.py

```diff
@@ -79,3 +78,69 @@ def __init__(self, opt: Dict[str, Any]):
             raise ValueError(
                 'There must be a "none_all_good" category in self.problem_buckets!'
             )
+
+
+# Defining the class only if Mephisto is installed, since it relies on Mephisto
+try:
+    from mephisto.abstractions.databases.local_database import LocalMephistoDB
+    from mephisto.data_model.unit import Unit
+    from mephisto.tools.data_browser import DataBrowser as MephistoDataBrowser
+
+    class AbstractDataBrowserResultsCompiler(AbstractResultsCompiler):
```

Comment:
if you drop the type annotations, or switch them to strings, the class definition doesn't need to be in the try

Reply:
Let's discuss structure before we merge this.

Reply:
Actually I think Python 3.7 solves it pretty easy for us. The following works fine:

```python
#!/usr/bin/env python3

from __future__ import annotations

try:
    from nonexistant import Nonimported
except ImportError:
    pass

class MyClass:
    def foo(self) -> Nonimported:
        pass
```